scrnaseq.Rmd

```diff
@@ -967,7 +1325,7 @@ loupe_genesets = rbind(loupe_genesets,
                                   Name=sc_markers_filt_down[,"gene"], 
                                   Ensembl=seurat_rowname_to_ensembl[sc_markers_filt_down[,"gene"]]))
 
-genesets_to_export = list(genes_cc_s_phase=genes_s, genes_cc_g2m_phase=genes_g2m)
+genesets_to_export = list(genes_cc_s_phase=genes_s[,2], genes_cc_g2m_phase=genes_g2m[,2])
```

Comment:
I previously had 3 lines of code or so where I restricted genes_s and genes_g2m to the genes present in the dataset, converting the two variables to character arrays.

Reply:
Where?

Reply:
Okay done

Reply:
See at the bottom: 

```
# Use biomart to translate human cell cycle genes to the species of interest
mart_human = biomaRt::useEnsembl("ensembl", dataset="hsapiens_gene_ensembl", mirror=mirror, version=param$ensembl_version) 
mart_myspecies = biomaRt::useEnsembl("ensembl", dataset=param$mart_dataset, mirror=mirror, version=param$ensembl_version) 

genes_s = biomaRt::getLDS(attributes=c("external_gene_name"), 
                          filters="external_gene_name", 
                          values=cc.genes.updated.2019$s.genes, 
                          mart=mart_human, 
                          attributesL=c("external_gene_name"), 
                          martL=mart_myspecies, 
                          uniqueRows=TRUE)
genes_g2m = biomaRt::getLDS(attributes=c("external_gene_name"), 
                            filters="external_gene_name", 
                            values=cc.genes.updated.2019$g2m.genes, 
                            mart=mart_human, 
                            attributesL=c("external_gene_name"), 
                            martL=mart_myspecies, 
                            uniqueRows=TRUE)

# Restrict genes to those present in the sc object
genes_s = genes_s[genes_s[,2] %in% rownames(sc), 2] 
genes_g2m = genes_g2m[genes_g2m[,2] %in% rownames(sc), 2]

```

Reply:
Done.